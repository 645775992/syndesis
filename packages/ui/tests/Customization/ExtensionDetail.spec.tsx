import * as React from 'react';
import { fireEvent, render } from 'react-testing-library';
import renderer from 'react-test-renderer';
import {
  ExtensionDetail,
  IExtensionDetailProps,
} from '../../src/Customization';

export default describe('ExtensionDetail', () => {
  const mockOnDelete = jest.fn();
  const mockOnUpdate = jest.fn();

  const cancelLabel = 'Cancel';
  const deleteLabel = 'Delete';
  const deleteModalMessage = 'Are you sure you want to delete the extension?';
  const deleteModalTitle = 'Confirm Delete?';
  const deleteTip = 'Delete this extension';
  const name = 'Loop';
  const extensionId = 'io.syndesis.extensions:syndesis-extension-loop';
  const idMsg = '(ID: ' + extensionId + ')';
  const overviewLabel = 'Overview';
  const supportedStepsText = 'Supported Steps';
  const updateLabel = 'Update';
  const updateTip = 'Update this extension';
  const usageLabel = 'Usage';
  const uses = 2;

  const props = {
    extensionName: name,
    extensionUses: uses,
    i18nCancelText: cancelLabel,
    i18nDelete: deleteLabel,
    i18nDeleteModalMessage: deleteModalMessage,
    i18nDeleteModalTitle: deleteModalTitle,
    i18nDeleteTip: deleteTip,
    i18nIdMessage: idMsg,
    i18nOverviewSectionTitle: overviewLabel,
    i18nSupportsSectionTitle: supportedStepsText,
    i18nUpdate: updateLabel,
    i18nUpdateTip: updateTip,
    i18nUsageSectionTitle: usageLabel,
    integrationsSection: <div />,
    onDelete: mockOnDelete,
    onUpdate: mockOnUpdate,
    overviewSection: <div />,
    supportsSection: <div />,
  } as IExtensionDetailProps;

  beforeEach(() => {
    mockOnDelete.mockReset();
    mockOnUpdate.mockReset();
  });

  it('Should render correctly', () => {
    const comp = <ExtensionDetail {...props} />;
    const { getByText, queryAllByText } = render(comp);

    // extension name
    expect(queryAllByText(name)).toHaveLength(1);

    // extension ID message
    expect(queryAllByText(idMsg)).toHaveLength(1);

    // update button
    expect(queryAllByText(updateLabel)).toHaveLength(1);
    fireEvent.click(getByText(updateLabel));
    expect(mockOnUpdate).toHaveBeenCalledTimes(1);

    // delete button
    expect(queryAllByText(deleteLabel)).toHaveLength(1);
    const deleteButton = getByText(deleteLabel);
    expect(deleteButton).toHaveAttribute('disabled'); // delete should be disabled

    // overview section title
    expect(queryAllByText(overviewLabel)).toHaveLength(1);

    // supports section title
    expect(queryAllByText(supportedStepsText)).toHaveLength(1);

    // usage section title
    expect(queryAllByText(usageLabel)).toHaveLength(1);

    // test snapshot
    const snapshot = renderer.create(comp).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('Should open delete confirmation modal', () => {
    const comp = <ExtensionDetail {...props} extensionUses={0} />;
    const { baseElement, getByText } = render(comp);
    const deleteButton = getByText(deleteLabel);
    expect(deleteButton).not.toHaveAttribute('disabled'); // delete should be enabled
    fireEvent.click(deleteButton); // open delete confirmation dialog
    expect(baseElement.classList).toContain('modal-open');

    // test snapshot
    const snapshot = renderer.create(comp).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
