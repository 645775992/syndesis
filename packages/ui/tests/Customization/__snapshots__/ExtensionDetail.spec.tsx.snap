// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtensionDetail Should open delete confirmation modal 1`] = `
<div
  className="card-pf card-pf-match-height"
>
  <div
    className="card-pf-heading"
  >
    <h2
      className="card-pf-title"
    >
      <div
        className="row"
      >
        <h1
          className="col-sm-8 extension-detail__extensionTitle"
        >
          Loop
          <span
            className="extension-detail__extensionId"
          >
            (ID: io.syndesis.extensions:syndesis-extension-loop)
          </span>
        </h1>
        <div
          className="col-sm-4 text-right extension-detail__titleButtons"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onBlur={[Function]}
            onClick={[MockFunction]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            Update
          </button>
          <button
            className="btn btn-default"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </h2>
  </div>
  <div
    className="card-pf-body"
  >
    <h3
      className="extension-detail__sectionHeading"
    >
      Overview
    </h3>
    <div />
    <h3
      className="extension-detail__sectionHeading"
    >
      Supported Steps
    </h3>
    <div />
    <h3
      className="extension-detail__sectionHeading"
    >
      Usage
    </h3>
    <div />
  </div>
</div>
`;

exports[`ExtensionDetail Should render correctly 1`] = `
<div
  className="card-pf card-pf-match-height"
>
  <div
    className="card-pf-heading"
  >
    <h2
      className="card-pf-title"
    >
      <div
        className="row"
      >
        <h1
          className="col-sm-8 extension-detail__extensionTitle"
        >
          Loop
          <span
            className="extension-detail__extensionId"
          >
            (ID: io.syndesis.extensions:syndesis-extension-loop)
          </span>
        </h1>
        <div
          className="col-sm-4 text-right extension-detail__titleButtons"
        >
          <button
            className="btn btn-primary"
            disabled={false}
            onBlur={[Function]}
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Class {
                      "_dispatchInstances": null,
                      "_dispatchListeners": null,
                      "_targetInst": null,
                      "altKey": null,
                      "bubbles": null,
                      "button": null,
                      "buttons": null,
                      "cancelable": null,
                      "clientX": null,
                      "clientY": null,
                      "ctrlKey": null,
                      "currentTarget": [Function],
                      "defaultPrevented": null,
                      "detail": null,
                      "dispatchConfig": null,
                      "eventPhase": null,
                      "getModifierState": [Function],
                      "isDefaultPrevented": [Function],
                      "isPropagationStopped": [Function],
                      "isTrusted": null,
                      "metaKey": null,
                      "movementX": [Function],
                      "movementY": [Function],
                      "nativeEvent": null,
                      "pageX": null,
                      "pageY": null,
                      "relatedTarget": [Function],
                      "screenX": null,
                      "screenY": null,
                      "shiftKey": null,
                      "target": null,
                      "timeStamp": [Function],
                      "type": null,
                      "view": null,
                    },
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            Update
          </button>
          <button
            className="btn btn-default"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            type="button"
          >
            Delete
          </button>
        </div>
      </div>
    </h2>
  </div>
  <div
    className="card-pf-body"
  >
    <h3
      className="extension-detail__sectionHeading"
    >
      Overview
    </h3>
    <div />
    <h3
      className="extension-detail__sectionHeading"
    >
      Supported Steps
    </h3>
    <div />
    <h3
      className="extension-detail__sectionHeading"
    >
      Usage
    </h3>
    <div />
  </div>
</div>
`;
